meta/store.go

```diff
@@ -1664,6 +1683,32 @@ func (fsm *storeFSM) Apply(l *raft.Log) interface{} {
 	return err
 }
 
+func (fsm *storeFSM) applyRemovePeerCommand(cmd *internal.Command) interface{} {
+	ext, _ := proto.GetExtension(cmd, internal.E_RemovePeerCommand_Command)
+	v := ext.(*internal.RemovePeerCommand)
+
+	id := v.GetID()
+	addr := v.GetAddr()
+	//Remove that node from the peer
+	if id == fsm.id {
+		// if we are part of the raft cluster, remove us
+		if _, ok := fsm.raftState.(*localRaft); ok {
+			fsm.Logger.Printf("removing peer for node id %d, %s", id, addr)
+			// ignore the error, as it's likely just telling you it's not the leader, and that's fine
```

Comment:
Shouldn't you explicitly test for this, and ignore if true? Is that possible?

This is your conscience speaking, not I. :-)